### PR TITLE
Dev tools guide updates

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Added a "Squash commits" section to the "Git commit message convention" guide.

Internal: Removed the `--debug` modifier from the "Running automated tests" section in the "Testing environment" guide as it is not supported after switching sources to TypeScript.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by psmyrek — DISMISSED** [review]
LGTM 🎉

File: docs/framework/contributing/git-commit-message-convention.md [hunk 1]
```diff
@@ -97,6 +97,12 @@ The proper order of sections for a commit message is as follows:
 
 All entries must be separated with a blank line, otherwise the lines will not be treated as separate entries.
 
+### Squash commits
+
+The changelog generator understands squash commits created by GitHub when merging a pull request.
+
+When using the _"Squash and merge"_ option, ensure the default commit title is not modified. It should contain the pull request title and its number, e.g., `Example pull request (#000)`. The changelog entries should be added as a commit description. They must follow the same rules as merge commits.
```
Comment (godai78) [inline docs/framework/contributing/git-commit-message-convention.md:104]:
```suggestion
When using the _"Squash and merge"_ option, ensure the default commit title is not modified. It should contain the pull request title and its number, for example: `Sample pull request (#000)`. The changelog entries should be added as a commit description. They must follow the same rules as merge commits.
```

**Review by godai78 — APPROVED** [review]

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Comment by pomek** [comment]
CI will likely fail due to selection changes in Chrome 129. See: #17113. As this PR updates only markdown files (docs), I will merge it anyway.

**Pull request merged**